package.py

```diff
@@ -1183,6 +1197,17 @@ def copy_file_to_target(file, temp_dir):
             # NOTE: poetry must be available in the build environment, which is the case with lambci/lambda:build-python* docker images but not public.ecr.aws/sam/build-python* docker images
             # FIXME: poetry install does not currently allow to specify the target directory so we export the
             # requirements then install them with "pip --no-deps" to avoid using pip dependency resolver
+
+            poetry_export = [
+                poetry_exec,
+                "export",
+                "--format",
+                "requirements.txt",
+                "--output",
+                "requirements.txt",
+                "--with-credentials",
+            ] + poetry_extra_args
```

Comment:
Sorry I did not notice this earlier, one additional remark: we may want to clarify that this only applies to the `poetry export` command with `poetry_export_extra_args`